qemu/tests/dump_guest_memory.py

```diff
@@ -0,0 +1,229 @@
+        """
+        Check if the expect key, val are contained in QMP output qmp_o.
+
+        :param qmp_o: output of QMP command
+        :type qmp_o: list
+        :param key: expect result
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: expect result is dict (True) or str (False)
+        :type check_item_in_pair: bool.
+
+        :return check result
+        :rtype: bool
+        """
+        for element in qmp_o:
+            if isinstance(element, dict):
+                if _check_dict(element, key, val, check_item_in_pair):
+                    return True
+            elif isinstance(element, list):
+                if check_list(element, key, val, check_item_in_pair):
+                    return True
+            elif element != '' and not check_item_in_pair:
+                if strict_match:
+                    if operator.eq(key, element):
+                        return True
+                else:
+                    if key in str(element):
+                        return True
+        return False
+
+    def _check_dict(dic, key, val, check_item_in_pair=True):
+        """
+        Check if the expect key, val are contained in QMP output dic.
+
+        :param dic: content of QMP command return value
+        :type dic: dict
+        :param key: expect result
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: If expect result is dict or str
+        :type check_item_in_pair: bool. Means expect result is dict or str.
+
+        :return check result
+        :rtype: bool
+        """
+        if key in dic and not check_item_in_pair:
+            return True
+        elif key in dic and val == dic[key]:
+            return True
+        else:
+            for value in dic.values():
+                if isinstance(value, dict):
+                    if _check_dict(value, key, val, check_item_in_pair):
+                        return True
+                elif isinstance(value, list):
+                    if check_list(value, key, val, check_item_in_pair):
+                        return True
+                elif value != '' and not check_item_in_pair:
+                    if strict_match:
+                        if operator.eq(key, value):
+                            return True
+                    else:
+                        if key in str(value):
+                            return True
+            return False
+
+    def check_result(qmp_o, expect_o=None):
+        """
+        Check test result with difference way according to result_check.
+        result_check = equal, expect_o should equal to qmp_o.
+        result_check = contain, expect_o should be contained in qmp_o
+        result_check = not_contain, expect_o should not be contained in qmp_o.
```

Comment:
There is no `not_contain` case and you actually does not handle it below, right?

Reply:
Yes,It originally existed, but it was deleted because it would not be used. I forgot to modify it here.
Will delete `result_check = not_contain, expect_o should not be contained in qmp_o.`